packages/auth/src/components/AuthorizationProvider.tsx

```diff
@@ -13,8 +15,10 @@ const AuthorizationProvider = ({ children }: AuthorizationProviderProps) => {
   const [{ data, fetching, stale, error }] = useGetCurrentAuthorizedUserQuery();
   const isLoaded = !fetching && !stale;
   let deleted = false;
+  const logger = useLogger();
 
-  if (error) {
+  if (error && containsUserDeletedError(error)) {
+    logger.debug("Detected UserDeleted error in AuthorizationProvider");
     deleted = true;
```

Comment:
This feels like it could be the core part of the problem, _any_ error sets `deleted = true` in the context that appears everywhere  
Hopefully this cleans things up nicely 🤞